src/easymotion/easymotion.ts

```diff
@@ -0,0 +1,408 @@
+  }
+
+  /**
+   * Create and cache decoration types for different marker lengths
+   */
+  public static getDecorationType(length: number): vscode.TextEditorDecorationType {
+    var cache = this.decorationTypeCache[length];
+    if (cache) {
+      return cache;
+    }
+
+    var width = length * 8;
+    var type = vscode.window.createTextEditorDecorationType({
+      after: {
+        margin: `0 0 0 -${width}px`,
+        height: `14px`,
+        width: `${width}px`
+      }
+    });
+
+    this.decorationTypeCache[length] = type;
+
+    return type;
+  }
+
+  /**
+   * Create and cache the SVG data URI for different marker codes and colors
+   */
+  private static getSvgDataUri(code: string, backgroundColor: string, fontColor: string): vscode.Uri {
+      var cache = this.svgCache[code];
+      if (cache) {
+        return cache;
+      }
+
+      const width = code.length * 8 + 1;
+      var uri = vscode.Uri.parse(
+        `data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 ${width} ` +
+        `13" height="14" width="${width}"><rect width="${width}" height="14" rx="2" ry="2" ` +
+        `style="fill: ${backgroundColor};"></rect><text font-family="Consolas" font-size="14px" ` +
+        `fill="${fontColor}" x="1" y="10">${code}</text></svg>`);
+
+      this.svgCache[code] = uri;
+
+      return uri;
+  }
+
+  /**
+   * Enter EasyMotion mode
+   */
+  public enterMode() {
+    this.accumulation = "";
+    this._vimState.currentMode = ModeName.EasyMotionMode;
+  }
+
+  /**
+   * Exit EasyMotion mode and clean up
+   */
+  public exitMode() {
+    this._vimState.currentMode = ModeName.Normal;
+
+    this.accumulation = "";
+    this.clearMarkers();
+    this.clearDecorations();
+  }
+
+  /**
+   * Clear all decorations
+   */
+  public clearDecorations() {
+    var editor = vscode.window.activeTextEditor;
+    for (var i = 1; i <= this.decorations.length; i++) {
+      editor.setDecorations(EasyMotion.getDecorationType(i), []);
+    }
+  }
+
+  /**
+   * Clear all markers
+   */
+  public clearMarkers() {
+    this.markers = [];
+    this.visibleMarkers = [];
+  }
+
+  public addMarker(marker: EasyMotion.Marker) {
+    this.markers.push(marker);
+  }
+
+  public getMarker(index: number): EasyMotion.Marker {
+    return this.markers[index];
+  }
+
+  /**
+   * Find markers beginning with a string
+   */
+  public findMarkers(nail: string, visible = true): EasyMotion.Marker[] {
+    var arr = visible ? this.visibleMarkers : this.markers;
+    var markers: EasyMotion.Marker[] = [];
+    for (var i = 0; i < arr.length; i++) {
+      var marker = arr[i];
+
+      if (marker.name.startsWith(nail)) {
+        markers.push(marker);
+      }
+    }
+
+    return markers;
+  }
+
+  /**
+   * Search and sort using the index of a match compared to the index of position (usually the cursor)
+   */
+  public sortedSearch(position: Position, searchString = "", options: EasyMotion.SearchOptions = {}): EasyMotion.Match[] {
```

Comment:
Instead of passing in `isRegex: true`, simply make the type of searchString `string | RegExp`. Then you can do `if (typeof searchString === "string")` and TypeScript is intelligent enough to infer that within the curly braces of that if, `searchString` is now a string (similarly with the RegExp).